src/main/java/com/gregtechceu/gtceu/api/data/chemical/material/info/MaterialFlag.java

```diff
@@ -45,6 +45,15 @@ protected Set<MaterialFlag> verifyFlag(Material material) {
         return thisAndDependencies;
     }
 
+    public Set<PropertyKey<?>> getAllRequiredProperties() {
+        Set<PropertyKey<?>> thisAndDeps = new HashSet<>(requiredProperties);
+        thisAndDeps.addAll(requiredFlags.stream()
+                .map(MaterialFlag::getAllRequiredProperties)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet()));
```

Comment:
```suggestion
		requiredFlags.stream()
                .map(MaterialFlag::getAllRequiredProperties)
                .forEach(thisAndDeps::addAll)
```
Change the code in the method above to match this as well - no need to flatmap multiple streams or have an unnecessary collection